### ISSUE TITLE
`typoTolerance.disableOnAttributes` does not seem to be working when `attributesToSearchOn` is used.

### ISSUE THREAD

**Issue by tobz-nz** [meilisearch/meilisearch#4232]
**Describe the bug**

The `typoTolerance.disableOnAttributes` seems to be ignored when the `attributesToSearchOn` search parameter is used.

**To Reproduce**
I've not done a separate test other than my current project as described in the **Additional context** section, but these steps should work?

1. create an index with several fields and disable typo tolerance on one of them, lets call it `exact_only`
2. fill the index with data, the `exact_only` field should have similar but not exact values.
3. do a search for a term found in the `exact_only` field and set `attributesToSearchOn=exact_only`
4. see more than 1 result.
5. the same search without `attributesToSearchOn` will only return 1 result as expected.

**Expected behavior**
The search with `attributesToSearchOn=exact_only` should only return 1 result - matching the exact search term, or those with the same term plus extra characters.
For example:

`q=test` should return results for `test` and `tested`, but not `tesd`

**Meilisearch version:**
1.5.0

**Additional context**
I've got an index of `products` that has some fields for the `gtin` number and supplier / manufacture product code's, which I have added to the `disableOnAttributes` array - but doing a search for a sku still returns other skus that are close to the search but not exact.

`/indexes/products/settings/typo-tolerance`
```json
{
  "enabled": true,
  "minWordSizeForTypos": {
    "oneTypo": 5,
    "twoTypos": 9
  },
  "disableOnWords": [],
  "disableOnAttributes": [
    "code",
    "gtin",
    "product",
    "supplier"
  ]
}
```

Simple search returns expected results:

`/indexes/local_products/search?q=BCK88C`
```json
{
  "hits": [
    {
      "title": "Brother LC431C Cyan Ink Cartridge",
      "code": "LC431C",
      "gtin": "4977766810616",
      "product": "DV15817",
      "supplier": "BCK88C",
      "id": 95884,
    }
  ],
  "query": "BCK88C",
  "processingTimeMs": 1,
  "limit": 20,
  "offset": 0,
  "estimatedTotalHits": 1
}
```

But with `attributesToSearchOn`, 14 results are returned - only 1 matches the `supplier` field.

`/indexes/local_products/search?attributesToSearchOn=code,supplier,product&q=BCK88C`
```json
{
  "hits": [
    {
      "title": "Brother LC431C Cyan Ink Cartridge",
      "code": "LC431C",
      "gtin": "4977766810616",
      "product": "DV15817",
      "supplier": "BCK88C",
      "id": 95884
    },
    {
      "title": "Brother LC39C Cyan Ink Cartridge",
      "code": "LC39C",
      "gtin": "4977766676731",
      "product": "dv3622",
      "supplier": "BCK18C",
      "id": 13384
    },
    {
      "title": "Brother LC23EC Cyan Super High Yield Ink Cartridge",
      "code": "LC23EC",
      "gtin": "4977766754958",
      "product": "DV09882",
      "supplier": "BCK82C",
      "id": 52932
    },
    ...
  ],
  "query": "BCK88C",
  "processingTimeMs": 2,
  "limit": 20,
  "offset": 0,
  "estimatedTotalHits": 14
}
```

**Comment by dureuill** [comment]
Hello @tobz-nz :wave:

thanks for the detailed report! This allowed @ManyTheFish to reproduce the issue.
We'll try to release a v1.5.1 with a fix when available.

**Comment by dureuill** [comment]
Fixed by #4233
